app/helpers/outlier.py

```diff
@@ -1,52 +1,94 @@
         self.outlier_dict["summary"] = textwrap.fill(outlier_summary, width=150)
+        self.doc = doc
+        self.cache_is_whitelist = None
 
     # Each whitelist item can contain multiple values to match across fields, separated with ",". So we need to
     # support this too.
     # Example: "dns_tunneling_fp = rule_updates.et.com, intel_server" -> should match both values across the entire
     # event (rule_updates.et.com and intel_server);
-    def is_whitelisted(self, additional_dict_values_to_check=None):
+    def is_whitelisted(self):
+        if self.cache_is_whitelist is None:
+            # Create dictionary that contain all stuff
+            self.cache_is_whitelist = Outlier.is_whitelisted_doc({'outlier_dict': self.outlier_dict,
+                                                                  'additional_dict_to_check': self.doc})
+        return self.cache_is_whitelist
+
+    def get_outlier_dict_of_arrays(self):
+        outlier_dict_of_arrays = dict()
+
+        for k, v in self.outlier_dict.items():
+            if isinstance(v, list):
+                outlier_dict_of_arrays[k] = v
+            else:
+                outlier_dict_of_arrays[k] = [v]
+
+        return outlier_dict_of_arrays
+
+    def __str__(self):
+        _str = "\n"
+        _str += "=======\n"
+        _str += "outlier\n"
+        _str += "=======\n"
+
+        for key, value in self.outlier_dict.items():
+            _str += (str(key) + "\t -> " + str(value) + "\n")
+
+        return _str
+
+    def __eq__(self, other):
+        return isinstance(other, Outlier) and self.outlier_dict == other.outlier_dict
+
+    @staticmethod
+    def is_whitelisted_doc(dict_to_check=None):
+        dict_values_to_check = set()
+
+        for dict_val in helpers.utils.nested_dict_values(dict_to_check):
+            if isinstance(dict_val, list):
+                for dict_val_item in dict_val:
+                    # force to be a string in case the nested element is a dictionary
+                    dict_values_to_check.add(str(dict_val_item))
+            else:
+                dict_values_to_check.add(dict_val)
+
         # Check if value is whitelisted as literal
         for (_, each_whitelist_configuration_file_value) in \
-                helpers.singletons.settings.config.items("whitelist_literals"):
+                helpers.singletons.settings.whitelist_literals_config:
             whitelist_values_to_check = each_whitelist_configuration_file_value.split(",")
 
             total_whitelisted_fields_to_match = len(whitelist_values_to_check)
             total_whitelisted_fields_matched = 0
 
             for whitelist_val_to_check in whitelist_values_to_check:
-                if self.matches_specific_whitelist_item_literally(whitelist_val_to_check,
-                                                                  additional_dict_values_to_check):
+                if Outlier.dictionary_matches_specific_whitelist_item_literally(whitelist_val_to_check,
```

Comment:
Seems like quite a huge method name

Reply:
agreed but it does explain what it does so I will keep it